Tools/Scripts/libraries/webkitapipy/webkitapipy/allow.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (C) 2025 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+
+import sys
+from dataclasses import dataclass, field
+from pathlib import Path
+from enum import Enum
+from typing import Any, Optional
+
+if sys.version_info < (3, 11):
+    from pip._vendor import tomli
+else:
+    import tomllib
+
+
+@dataclass
+class AllowedSPI:
+    key: str
+    bug: str | PermanentlyAllowedReason
+
+    symbols: list[str]
+    selectors: list[str]
+    classes: list[str]
+
+
+class PermanentlyAllowedReason(Enum):
```

Comment:
Maybe add a comment here saying we should just use `enum.StrEnum` once we require Python 3.11? Or even do that now, on Python 3.11 and later.

Reply:
I read the docs a bit closer, which suggest creating a StrEnum class on pre-3.11 pythons like this:

```python
class StrEnum(str, Enum):
    pass
```

which does indeed seem to work.